lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -548,16 +548,20 @@ def run_config_commands
     end
 
     def run_pre_start_commands
-      @pre_start_commands.each do |dirty_cmd|
-        cmd = replace_string_recursive(dirty_cmd, full_replacement_list)
-        run_cmd(cmd)
+      Timeout.timeout(60) do
```

Comment:
Timeout should be performed in the main thread.

Reply:
But run_pre_start_commands is run in a separate thread in the loop. How do we perform timeout from the main thread?

Reply:
Why we can't just fail qemu_thread?

Reply:
> timeout around `AutoHCK::HCKTest::run_and_configure_setup`

This is simple from the code view but difficult to select the proper timeout. `configure_setup` takes about 3-5 min with my NVME SSD but takes about 10-15 with HDD. Also on one Windows insider preview version, I see about 20 min initialization (not hangs on during `Waiting for client #{@name} initialization`). 

Currently, we have 30 min timeout in `AutoHCK::HCKClient::synchronize` (`@cooldown_thread&.join(CLIENT_COOLDOWN_TIMEOUT)`). So timeout around `AutoHCK::HCKTest::run_and_configure_setup` should be the biggest. But this is a bad situation to wait 45 minutes because QEMU can't start.

I think we need some local timeout near QEMU run.

Reply:
I see. Let's try the second option then; add code to deliver an exception to the thread spawned `qemu_thread`.

Reply:
Exceptions in `qemu_thread` are automatically delivered to the main thread due to `Thread.abort_on_exception` in `bin/auto_hck`.